### PR TITLE
python: Assorted D-Bus fixes

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] #17739 for the testing groundwork

### CODE REVIEW

File: src/cockpit/channels/dbus.py [hunk 1]
```diff
@@ -243,7 +244,9 @@ async def do_call(self, call, message):
             # watch processing, wait for that to be done.
             async with self.watch_processing_lock:
                 # TODO: stop hard-coding the endian flag here.
-                self.send_message(reply=[reply], id=cookie, flags="<" if flags is not None else None)
+                self.send_message(reply=[reply], id=cookie,
+                                  flags="<" if flags is not None else None,
+                                  type=type if type is not None else None)
```
Comment (allisonkarlitskaya) [inline src/cockpit/channels/dbus.py:256]: `type if type is not None else None` is a very fancy way to say `type`.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
One very minor issue.  Otherwise looks good.

File: src/cockpit/channels/dbus.py (hunk 1 above)
Comment (mvollmer) [reply src/cockpit/channels/dbus.py]: Indeed!  Fixed.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]
2022-12-13T08:01:28.9016842Z not ok 2 /static-code/test-flake8
2022-12-13T08:01:28.9020933Z # src/cockpit/channels/dbus.py:239:13: F841 local variable 'error' is assigned to but never used

**Review by allisonkarlitskaya — APPROVED** [review]
:+1:

**Comment by mvollmer** [comment]
Test failure was caused by cockpit-pcp segfaulting, probably out of fear of being rewritten in Python next.

**Pull request merged**